File: solutions/python/bob/bob.py
def response(hey):
    hey = hey.strip()
    if not hey:
        return "Fine. Be that way!"
    if hey.endswith("?"):
        if hey.isupper():
            return "Calm down, I know what I'm doing!"
        return "Sure."
    if hey.isupper():
        return "Whoa, chill out!"
    return "Whatever."

File: solutions/ruby/bob/bob.rb
# frozen_string_literal: true

module Bob
  def self.hey(sentence)
    sentence.strip!
    yelling = sentence == sentence.upcase && sentence != sentence.downcase
    return "Calm down, I know what I'm doing!" if yelling && sentence.end_with?('?')
    return 'Sure.' if sentence.end_with?('?')
    return 'Whoa, chill out!' if yelling
    return 'Fine. Be that way!' if sentence.empty?

    'Whatever.'
  end
end
